### ISSUE TITLE
affected is broken due to incorrect gitignore handling

### ISSUE THREAD

**Issue by cdaringe** [nrwl/nx#27888]
### Current Behavior

`affected -t build --files=packages/foo/src/index.ts` yields no output

### Expected Behavior

The above command should yield package `foo`

### GitHub Repo

n/a

### Steps to Reproduce

n/a

You can reproduce, but see below conversation. This behavior is traced.

### Nx Report

```shell
19.7.2
```

### Failure Logs

```shell
n/a
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Here's the scoop.

Suppose you have two types of packages:

- package type `foo`, e.g. a generic node lib
- package type `bar`

The `{workspaceRoot}/.gitignore` has:

```
# {workspaceRoot}/.gitignore
src/**
```

...in it, because 99% of your packages are `bar`s. Imagine that `bar` packages are all codegen'd, so you don't want to track `src`. It seemed easier to to put this in the _root_ ignore vs every per-package gitignore.

Now, imagine that `packages/foo` undoes that root gitignore with:

```
# {workspaceRoot}/packages/foo/.gitignore
!src/**
```

Now, try doing `affected --files=packages/foo/src/some/modue.ts`. Bummer, affected says that `foo` is not affected!

TLDR: https://github.com/nrwl/nx/blob/b6140d459020847818b738ddc17e52c069657893/packages/nx/src/project-graph/file-utils.ts#L67C12-L67C27 **does not consider package gitignores**, thus incorrectly computes affected.

Discussion:

- affected should pick up on project/package level gitignores, or
- error on their presence, as they lead to surprising behavior
